src/api.rs

```diff
@@ -54,15 +54,17 @@ impl Config {
         aom_dsp_rtcd();
     }
 
-    Context { fi, seq, frame_q: VecDeque::new() }
+    Context { fi, seq, frame_count: 0, idx: 0, frame_q: VecDeque::new() }
   }
 }
 
 pub struct Context {
   fi: FrameInvariants,
   seq: Sequence,
   //    timebase: Ratio,
-  frame_q: VecDeque<Option<Arc<Frame>>> //    packet_q: VecDeque<Packet>
+  frame_count: u64,
+  idx: u64,
+  frame_q: VecDeque<(u64, Option<Arc<Frame>>)> //    packet_q: VecDeque<Packet>
```

Comment:
You might want to use a map probably e.g. [btreemap](https://doc.rust-lang.org/stable/std/collections/struct.BTreeMap.html).

Reply:
You are right. A BTreeMap is much cleaner. Thanks for the tip!